sphinx/util/console.py

```diff
@@ -22,21 +23,8 @@ def terminal_safe(s: str) -> str:
 
 
 def get_terminal_width() -> int:
-    """Borrowed from the py lib."""
-    if sys.platform == "win32":
-        # For static typing, as fcntl & termios never exist on Windows.
-        return int(os.environ.get('COLUMNS', 80)) - 1
-    try:
-        import fcntl
-        import struct
-        import termios
-        call = fcntl.ioctl(0, termios.TIOCGWINSZ, struct.pack('hhhh', 0, 0, 0, 0))
-        height, width = struct.unpack('hhhh', call)[:2]
-        terminal_width = width
-    except Exception:
-        # FALLBACK
-        terminal_width = int(os.environ.get('COLUMNS', 80)) - 1
-    return terminal_width
+    """Get number of columns of the terminal."""
+    return shutil.get_terminal_size().columns - 1
```

Comment:
Previously we only checked COLUMNS if querying the terminal directly failed--perhaps we should maintain that order of operations?:

```suggestion
    """Return the width of the terminal window.

    The terminal connected to sys.__stdout__ is queried by invoking
    os.get_terminal_size. If the terminal size cannot be determined, the
    environment variable COLUMNS is used if defined. Otherwise, fallback value
    is returned.
    """
    try:
        return os.get_terminal_size(sys.__stdout__.fileno()).columns - 1
    except (AttributeError, ValueError, OSError):
        # FALLBACK
        return int(os.environ.get('COLUMNS', 80)) - 1
```

A

Reply:
Good catch. On the other hand - the order of precedence implemented by `shutil.get_terminal_size` is not without reason. If the user wants (for whatever reason, admittedly I can't think of any right now) to restrict the output width she or he can do it by defining `COLUMNS`. In the current program flow, there is no possibility to do so. So it's a question of backward compatibility for Linux systems (Windows was fixed at 80) versus the principle of least surprise (is someone goes to the trouble to define the environment variable one can assume it's done for a good reason). So I'm neural on it.